ZipBuilder/Sources/ZipBuilder/FirebaseBuilder.swift

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright 2019 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+
+/// Wrapper for the Firebase zip build.  Unlike the generic zip builder, the Firebase build creates a two-level
+/// zip with the option to install different Firebase library subsets.
+struct FirebaseBuilder {
+  /// ZipBuilder instance.
+  private let zipBuilder: ZipBuilder
+
+  /// Default initializer. If allSDKsPath and currentReleasePath are provided, it will also verify
+  /// that the
+  ///
+  /// - Parameters:
+  ///   - paths: Paths that are needed throughout the process of packaging the Zip file.
+  ///   - customSpecRepo: A custom spec repo to be used for fetching CocoaPods from.
+  init(zipBuilder: ZipBuilder) {
+    self.zipBuilder = zipBuilder
+  }
+
+  func build(in projectDir: URL) {
```

Comment:
Nit: Please add a documentation comment to this function.

Reply:
Done.